### PR TITLE
GEN-39: Update Renovate config

### PR DESCRIPTION
**Pull request by vilkinsons**
## 🌟 What is the purpose of this PR?

Updates our Renovate configuration file.

Additionally makes drive-by changes to:

- the roadmap page we link to from the root README, as well, to direct GtiHub users to our development roadmap at [hash.dev/roadmap](https://hash.dev/roadmap) over the current (and less up-to-date) [hash.ai/roadmap](https://hash.ai/roadmap) page
- the `libs` README's description of our _internal libraries_ published to npm.

## 🔗 Related links

- Renovate's docs detailing its [Configuration Options](https://docs.renovatebot.com/configuration-options/)

## 🚫 Blocked by

- #2891 merging. We will try re-running the CI lint fix script once it's in, using this PR as the test one for that.

## 🔍 What does this change?

- **Branch prefixes:** renovate shortened to deps, and env-specific prefixes appended
- **Groups:** new groups added

Previously had intended to add self-labeling here, but we have decided to keep this centralized in the `labeler.yml` file to avoid splitting where this occurs.

### 🚢 Has this modified a publishable library?

- [x] **No.** This PR does not modify any publishable blocks or libraries, or modifications do not need publishing

### 📜 Does this require a change to the docs?

- [x] **No.** This PR is internal and does not require a docs change

### 🕸️ Does this require a change to the Turbo Graph?

- [x] **I am unsure and need a bit of an explainer.** I doubt it does, but I've never worked with the Turbo Graph before and don't know how it works.

## ⚠️ Known issues

- None ✅

## 🐾 Next steps

- Ideally find a way to test this out before merging, or subject it to proper code review.

## 🛡 What tests cover this?

- None that I am aware of.

## ❓ How to test this?

- Unsure how we can without merging to `main` 🫣

## 📹 Demo

- _n/a_

### CODE REVIEW

File: .github/renovate.json [hunk 1]
```diff
@@ -11,6 +13,24 @@
...
+      "groupName": "Block Protocol",
+      "matchManagers": ["npm"],
+      "matchPackagePatterns": [
+        "^@blockprotocol/",
+        "^block-",
+        "blockprotocol",
+        "create-block-app",
+        "mock-block-dock"
+      ],
+      "addLabels": "area: blocks"
```
Comment (TimDiekmann) [inline .github/renovate.json:32]: Isn't this label added by the labeler flow anyway? I'd rather keep it in one location.
```suggestion
      ]
```

Comment (nonparibus) [reply .github/renovate.json:32]: Let me confirm. Stupidly didn't really think about the `labeler` as I implemented this, but I 100% agree it would be best to have a single source of truth.

Comment (TimDiekmann) [reply .github/renovate.json:32]: The list is not as big as your new list but we can update that instead:
https://github.com/hashintel/hash/blob/12fecc40c71cf21350af50e198d58c8baadaadfc/.github/labeler.yml#L27-L28

**Comment by TimDiekmann** [comment]
> Unsure how we can without merging to main 🫣

Not worth it. It's possible by forking it and setting up renovate for the fork. I wouldn't do it.

File: .github/renovate.json (hunk 1 above)
Comment (nonparibus) [reply .github/renovate.json:32]: @TimDiekmann I agree with your principle of removing all the `addLabels` from this PR, where we can achieve the same thing through better rules in `labeler.yml`. Would you mind taking the sister task at some point in the future? Captured as GEN-40 (extremely low prio... I was only picking this low-prio ticket up in the first place as I needed to clear my mind with something mundane for ~20 before getting back to writing).

Comment (TimDiekmann) [reply .github/renovate.json]: Sure! I added a back-link to this issue so I'm able to find it again when picking it up.

File: .github/renovate.json [hunk 2]
```diff
@@ -20,35 +38,72 @@
         "^@typescript-eslint/",
         "^eslint-",
         "^eslint$"
-      ]
+      ],
```
Comment (TimDiekmann) [inline .github/renovate.json:41]:
```suggestion
      ]
```

File: .github/renovate.json [hunk 3]
```diff
@@ -20,35 +38,72 @@
...
+      "groupName": "GraphQL",
+      "matchManagers": ["npm"],
+      "matchPackagePatterns": ["^@graphql/", "^graphql$", "^graphql-"]
     },
     {
       "enabled": true,
       "groupName": "Jest",
       "matchManagers": ["npm"],
-      "matchPackagePatterns": ["^@types/jest$", "^jest", "^ts-jest$", "^jest-"]
+      "matchPackagePatterns": ["^@types/jest$", "jest", "^jest-", "-jest$"],
```
Comment (TimDiekmann) [inline .github/renovate.json:53]:
```suggestion
      "matchPackagePatterns": ["^@types/jest$", "jest", "^jest-", "-jest$"]
```

File: .github/renovate.json [hunk 4]
```diff
@@ -20,35 +38,72 @@
...
+      "groupName": "OpenTelemetry",
+      "matchManagers": ["npm"],
+      "matchPackagePatterns": ["^@opentelemetry/"]
     },
     {
       "enabled": true,
       "groupName": "Playwright",
       "matchManagers": ["npm"],
-      "matchPackagePatterns": ["^@playwright/", "^playwright$", "^playwright-"]
+      "matchPackagePatterns": ["^@playwright/", "^playwright$", "^playwright-"],
```
Comment (TimDiekmann) [inline .github/renovate.json:65]:
```suggestion
      "matchPackagePatterns": ["^@playwright/", "^playwright$", "^playwright-"]
```

File: .github/renovate.json [hunk 5]
```diff
@@ -20,35 +38,72 @@
...
+    },
+    {
+      "enabled": true,
+      "groupName": "Prettier",
+      "matchManagers": ["npm"],
+      "matchPackagePatterns": ["^prettier"],
```
Comment (TimDiekmann) [inline .github/renovate.json:71]:
```suggestion
      "matchPackagePatterns": ["^prettier"]
```

File: .github/renovate.json [hunk 6]
```diff
@@ -20,35 +38,72 @@
...
     },
     {
       "enabled": true,
       "groupName": "ProseMirror",
       "matchManagers": ["npm"],
-      "matchPackagePatterns": ["^prosemirror-"]
+      "matchPackagePatterns": ["^prosemirror$"]
```
Comment (TimDiekmann) [inline .github/renovate.json:77]: Was this change intentional or did you revert the wrong pattern?

File: .github/renovate.json [hunk 7]
```diff
@@ -22,33 +40,70 @@
...
+      "groupName": "GraphQL",
+      "matchManagers": ["npm"],
+      "matchPackagePatterns": ["^@graphql/", "^graphql$", "^graphql-"]
+    },
     {
       "enabled": true,
       "groupName": "Jest",
       "matchManagers": ["npm"],
-      "matchPackagePatterns": ["^@types/jest$", "^jest", "^ts-jest$", "^jest-"]
+      "matchPackagePatterns": ["^@types/jest$", "jest", "^jest-", "-jest$"]
```
Comment (TimDiekmann) [inline .github/renovate.json:53]: What was the reason for this change? `"jset"` is a very generic pattern (which will actually cover all the other patterns in this line as well).

Comment (nonparibus) [reply .github/renovate.json]: I was hoping to avoid potential false positives, but am realizing that the logic here could be simplified in any case.

And actually, upon further reflection, if we end up with false positives we can just refine it then. Thanks for flagging. Amended in https://github.com/hashintel/hash/commit/198b1c5e5b7dcecfb70d12e7a06d4661d1850708

File: .github/renovate.json (hunk 6 above)
Comment (nonparibus) [reply .github/renovate.json:77]: We have some packages that end in `prosemirror`, as well as start, so I thought this would be an improvement.

File: .github/renovate.json [hunk 8]
```diff
@@ -22,33 +40,70 @@
         "^eslint$"
       ]
     },
+    {
+      "enabled": true,
+      "groupName": "GraphQL",
+      "matchManagers": ["npm"],
+      "matchPackagePatterns": ["^@graphql/", "^graphql$", "^graphql-"]
```
Comment (nonparibus) [inline .github/renovate.json:47]:
```suggestion
      "matchPackagePatterns": ["^@graphql/", "^graphql$"]
```

It's redundant, right? @TimDiekmann

File: .github/renovate.json (hunk 7 above)
Comment (TimDiekmann) [reply .github/renovate.json]: I'm pretty sure the amended version is not what we want. With this we won't capture `@types/jest`, `ts-jest`, `eslint-plugin-jest`, `babel-jest`, `jest-junit`, `jest-message-util` (various others).
```suggestion
"matchPackagePatterns": ["^@types/jest$", "^jest-", "-jest$"]
```

Comment (nonparibus) [reply .github/renovate.json]: My bad. Actually fixed, not broken more, in https://github.com/hashintel/hash/pull/2887/commits/cb8bc7a73ef70dc0cedfdde7071cd6e2413cbaf0

**Review by TimDiekmann — APPROVED** [review]
LGTM, let's test it soon-ish 🙂

File: .github/renovate.json [hunk 9]
```diff
@@ -19,36 +37,90 @@
...
+      "matchManagers": ["poetry"],
+      "commitMessageTopic": "Python package `{{depName}}`",
+      "additionalBranchPrefix": "py/"
+    },
+    {
+      "enabled": true,
+      "groupName": "Pydantic",
+      "matchManagers": ["poetry"],
+      "matchPackagePatterns": ["^pydantic"]
+    },
```
Comment (nonparibus) [inline .github/renovate.json:118]: @TimDiekmann Could I have combined these into one? I think possibly yes, right?

Comment (TimDiekmann) [reply .github/renovate.json]: Which one do you mean, the python related groups? I don't see how we could combine them as you add a group name for `Pydantic`. I don't know if we need that, though 🤷🏻‍♂️

**Comment by nonparibus** [comment]
@TimDiekmann For testing:
- opening a Python PR to test branch prefixing, team assignment and commit/PR titling 
- opening a Rust PR to test branch prefixing and team assignment
- opening a JS/TS PR to test branch prefixing (could do one or more of new/changed groups specifically: `Block Protocol`, `ESLint`, `GraphQL`, `Jest`, `Playwright`, `Prettier`, `ProseMirror` and `Signia`, although we'll see their groupings in the dependency dashboard anyway)

File: .github/renovate.json (hunk 9 above)
Comment (nonparibus) [reply .github/renovate.json]: My brain is literally not working this morning. I'm going to take a nap.

**Review by TimDiekmann — APPROVED** [review]

**Review by TimDiekmann — DISMISSED** [review]

File: .github/renovate.json [hunk 10]
```diff
@@ -11,6 +12,23 @@
...
+      "groupName": "Block Protocol",
+      "matchManagers": ["npm"],
+      "matchPackagePatterns": [
+        "^@blockprotocol/",
+        "^block-",
+        "^blockprotocol$",
+        "^create-block-app$",
+        "^mock-block-dock$"
+      ]
+    },
```
Comment (TimDiekmann) [inline .github/renovate.json:31]: I just noted that `groupName` does not mean the packages are put in the same section but a PR for all these files will be created at once. In this case this is likely not what as we for example don't want to update all `@blockprotocol` at once.
```suggestion
```

Comment (TimDiekmann) [reply .github/renovate.json:31]: The PR is still approved, just hit `merge` if you disagree with me 🙂

Comment (nonparibus) [reply .github/renovate.json:31]: Why wouldn't we want to merge all Þ packages imported into the # repo at once? cc @CiaranMn

Comment (CiaranMn) [reply .github/renovate.json:33]: If we keep on top of it, it should be fine to upgrade all BP packages at once, because they're mostly interconnected, and new releases of some will often just be upgrading the version of another one, in which case it makes sense to do them together.

If there are releases of independent packages (e.g. `@blockprotocol/hook` and `@blockprotocol/events`) then it becomes a pain because you have to do unrelated work in the same PR to fix breaking changes or whatever is required, and you might end up splitting it out.

I don't think it particularly matters either way. We can pick one approach and change it later if it does.

Comment (TimDiekmann) [reply .github/renovate.json:33]: This means, that we cannot use renovate anymore to bump the type-system package as this would require updating all other BP packages. Yes, most packages are connected, but not all.

File: .github/renovate.json [hunk 11]
```diff
@@ -19,36 +39,74 @@
...
+      "groupName": "Pydantic",
+      "matchManagers": ["poetry"],
+      "matchPackagePatterns": ["^pydantic"]
+    },
     {
       "matchManagers": ["cargo"],
-      "commitMessageTopic": "Rust crate `{{depName}}`"
+      "commitMessageTopic": "Rust crate `{{depName}}`",
+      "additionalBranchPrefix": "rs/",
+      "reviewers": ["team:Rust"]
```
Comment (TimDiekmann) [inline .github/renovate.json:109]: Do we want to add this to the toolchain groups as well?

Comment (TimDiekmann) [reply .github/renovate.json:109]: (The branch prefix and the reviewers)

Comment (nonparibus) [reply .github/renovate.json:108]: @TimDiekmann I think this is done in https://github.com/hashintel/hash/commit/b8d865a7105bd4cb721672d9d6ad3b2321a6a4af -- if you could check I've caught all Rust crates as well as the toolchain updates which use the custom regex manager approach that'd be great.

Comment (TimDiekmann) [reply .github/renovate.json:108]: Looks good, let's try it 🙂

**Review by TimDiekmann — APPROVED** [review]

**Comment by nonparibus** [comment]
@TimDiekmann I'm re-running lint fix on this now ([job](https://github.com/hashintel/hash/actions/runs/5855308430/job/15872853929)). Will add to the merge queue when done.

File: .github/renovate.json [hunk 12]
```diff
@@ -57,6 +118,8 @@
       "separateMajorMinor": false,
       "fetchReleaseNotes": false,
       "commitMessageExtra": "to nightly-{{newVersion}}"
```
Comment (TimDiekmann) [inline .github/renovate.json:120]:
```suggestion
      "commitMessageExtra": "to nightly-{{newVersion}}",
```

**Review by TimDiekmann — DISMISSED** [review]

**Review by TimDiekmann — APPROVED** [review]
